### PR TITLE
tdesktop: 3.1.11 -> 3.4.3

### PR DESCRIPTION
**Pull request by MrMebelMan**
###### Motivation for this change
Update to a new version

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by MrMebelMan** [comment]
image
I've encountered this issue while trying to test the new version, but I'm not sure what to do about it 👉👈

**Comment by Kranzes** [comment]
TDesktop now depends on qt6, which is not available in nixpkgs currently.
See: https://github.com/NixOS/nixpkgs/pull/141883

**Comment by VergeDX** [comment]
Yes, but we can disabled the Qt6 feature by adding `"-DDESKTOP_APP_QT6=OFF"` to `cmakeFlags`.
I tried to update and meets some issues, I will checkout the update tomorrow : )

**Review by VergeDX — CHANGES_REQUESTED** [review]
Try to add `"-DDESKTOP_APP_QT6=OFF"` in `cmakeFlags`.

**Comment by VergeDX** [comment]
3.4.0 is released, I quickly tried it and seems everything is ok.
Please update to the latest, and skip qt6 requirement while building.

**Comment by MrMebelMan** [comment]
Updated to 3.4.0. Everything seems to work well, thanks @Kranzes & @VergeDX :heart:

**Review by VergeDX — APPROVED** [review]

File: pkgs/applications/networking/instant-messengers/telegram/tdesktop/default.nix [hunk 1]
```diff
@@ -66,7 +66,7 @@ let
 in
 mkDerivation rec {
   pname = "telegram-desktop";
-  version = "3.1.11";
+  version = "3.4.0";
```
Comment (VergeDX) [inline pkgs/applications/networking/instant-messengers/telegram/tdesktop/default.nix:69]: 3.4.1 just release, that was fast xD.

**Review by VergeDX — CHANGES_REQUESTED** [review]

**Comment by Kranzes** [comment]
Add a TODO comment above where you disabled QT6, something like "Remove once QT6 is available in nixpkgs".

**Comment by MrMebelMan** [comment]
Jesus

File: pkgs/applications/networking/instant-messengers/telegram/tdesktop/default.nix [hunk 2]
```diff
@@ -75,7 +75,7 @@ mkDerivation rec {
     repo = "tdesktop";
     rev = "v${version}";
     fetchSubmodules = true;
-    sha256 = "0hpcsraw3gx37wk3r88q51qf68ny0kb3kd6slnklwf22d1jqr3fn";
+    sha256 = "1397rsb1srzzfvbbqafhpf1jpx8iz13vj8y6gyq44gr9zq54iwcm";
```
Comment (1000101) [inline pkgs/applications/networking/instant-messengers/telegram/tdesktop/default.nix:78]:
```suggestion
    sha256 = "0sgfhicb81a5cn05f9zi7333kwh76qbnija1fyw0cxrrwap9d73m";
```
Sirs, this is `3.4.0`'s hash.

Comment (1000101) [inline pkgs/applications/networking/instant-messengers/telegram/tdesktop/default.nix:78]:
```suggestion
    sha256 = "1397rsb1srzzfvbbqafhpf1jpx8iz13vj8y6gyq44gr9zq54iwcm";
```
Sirs, this is 3.4.0's hash.

Comment (1000101) [inline pkgs/applications/networking/instant-messengers/telegram/tdesktop/default.nix:78]:
```suggestion
    sha256 = "0sgfhicb81a5cn05f9zi7333kwh76qbnija1fyw0cxrrwap9d73m";
```
Sirs, this is 3.4.0's hash.

Comment (MrMebelMan) [reply pkgs/applications/networking/instant-messengers/telegram/tdesktop/default.nix:78]: oh no

**Comment by 1000101** [comment]
Result of `nixpkgs-review pr 152702` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tdesktop</li>
  </ul>
</details>

**Comment by MrMebelMan** [comment]
> IT WOKZ
> 
> Result of `nixpkgs-review pr 152702` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 1 package built:

DIKI

**Comment by MrMebelMan** [comment]
Happy COVID-22, sirs! :tada:

**Comment by VergeDX** [comment]
3.4.2 plz xD

**Comment by 1000101** [comment]
> 3.4.2 plz xD

Now that's some agile development!

3.4.1:
>     Bug fixes and other minor improvements.

3.4.2:
>     Bug fixes and other minor improvements.

3.4.3:
WEN

**Comment by MrMebelMan** [comment]
WHY ARE YOU DOING THIS TO ME

**Comment by 1000101** [comment]
> WHY ARE YOU DOING THIS TO ME

@MrMebelMan commit it before it lays more versions

**Comment by MrMebelMan** [comment]
shouldnt

**Comment by 1000101** [comment]
Result of `nixpkgs-review pr 152702` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tdesktop</li>
  </ul>
</details>

**Review by VergeDX — APPROVED** [review]

**Review by 1000101 — APPROVED** [review]

**Comment by VergeDX** [comment]
The aarch64 build failed with internal complier error (ICE).
Idk what happened, I'll check out this tomorrow : (

**Comment by VergeDX** [comment]
The default cc of aarch64 is gcc9 (x86_64 on is gcc10), which caused the ICE.
I switch to gcc10Stdenv and compile it in arm64 Drone CI, it seems the error disappeared.

The commit: https://github.com/VergeDX/nixpkgs/commit/687d6cc5da37725834991c268ea2d7523140d401#diff-a45a9e94415ea5231fbac619e3eca6614b75e730f93f5986c163fc9304242517R69
The CI log: https://cloud.drone.io/VergeDX/actions-nix/9/1/2

**Comment by 1000101** [comment]
@GrahamcOfBorg build tdesktop

**Comment by ajs124** [comment]
> The default cc of aarch64 is gcc9 (x86_64 on is gcc10), which caused the ICE. I switch to gcc10Stdenv and compile it in arm64 Drone CI, it seems the error disappeared.

For reference #108305

**Comment by MrMebelMan** [comment]
Thanks @VergeDX, u da boss

**Review by 1000101 — APPROVED** [review]

**Comment by 1000101** [comment]
Merge it before OH NOES https://github.com/telegramdesktop/tdesktop/releases/tag/v3.4.3

File: pkgs/applications/networking/instant-messengers/telegram/tdesktop/default.nix [hunk 3]
```diff
@@ -63,10 +65,12 @@ let
       cxxStandard = "17";
     };
   };
+
```
Comment (VergeDX) [inline pkgs/applications/networking/instant-messengers/telegram/tdesktop/default.nix:68]: Maybe we can add a comment here:
`Aarch64 default gcc9 will cause ICE. For reference #108305`

**Review by VergeDX — CHANGES_REQUESTED** [review]

**Comment by MrMebelMan** [comment]
We need to hurry, comerades - 3.4.4 is brewing image

**Review by VergeDX — APPROVED** [review]

**Comment by 1000101** [comment]
Result of `nixpkgs-review pr 152702` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tdesktop</li>
  </ul>
</details>

**Review by 1000101 — APPROVED** [review]

**Pull request merged**